modules/products/products.service.js

```diff
@@ -0,0 +1,77 @@
+const Products = require('./products.model');
+const Size = require('../../models/Size');
+
+class ProductsService {
+  getAllProducts() {
+    return Products.find();
+  }
+
+  getProductsById(id) {
+    return Products.findById(id);
+  }
+
+  getSizeById(id) {
+    return Size.findById(id);
+  }
+
+  filterItems(args = {}){
+    const filter = {};
+    const {
+      pattern, colors, price
+    } = args;
+  
+    if (colors.length) {
+      filter.colors = {
+        $elemMatch: {
+          simpleName: { $in: colors },
+        },
+      };
+    }
+    if (pattern.length) {
+      filter.pattern = {
+        $elemMatch: {
+          value: { $in: pattern },
+        },
+      };
+    }
+    if(price){
+      filter.basePrice = {
+          $gte: price[0], 
```

Comment:
It would be a good idea to add validation that price is a positive number

Reply:
changed it to "price.length" because it is an array with max and min value

Reply:
made default value price = [0, 999999], and frontend part has min value 0